### PR TITLE
Adding pointwise linear block as an alternative to 1x1 Conv.

### LINKED ISSUE #631: Grad strides mismatch warning during DDP backward pass
🤖

## Problem

During the first backward pass, PyTorch's DDP reducer logs a warning about gradient stride mismatch:

```
Grad strides do not match bucket view strides. This may indicate grad was not created
according to the gradient layout contract, or that the param's strides changed since
DDP was constructed. This is not an error, but may impair performance.
grad.sizes() = [320, 256, 1, 1], strides() = [256, 1, 256, 256]
bucket_view.sizes() = [320, 256, 1, 1], strides() = [256, 1, 1, 1]
```

The gradient has strides `[256, 1, 256, 256]` while DDP expects `[256, 1, 1, 1]`. This forces DDP to copy gradients into contiguous buckets rather than using views, which adds overhead to every gradient synchronization.

## Context

This likely comes from a Conv2d layer with kernel size 1×1 where the gradient layout doesn't match what DDP expects. The parameter shape `[320, 256, 1, 1]` suggests a pointwise convolution, possibly in the FOMO encoder/decoder or ConvNeXt blocks.

## Potential approaches

- Identify which layer(s) produce non-contiguous gradients
- Consider calling `.contiguous()` on the parameter or restructuring the layer to avoid the stride mismatch
- Alternatively, this may be acceptable if the copy overhead is small relative to the gradient sync

### PR DESCRIPTION
**Pull request by alxmrs**
Fixes #631. Also, this more closely mimics the official implementation of ConvNext: https://github.com/facebookresearch/ConvNeXt/blob/main/models/convnext.py#L18.

### CODE REVIEW

**Comment by alxmrs** [comment]
@codex may I have your review?

**Comment by alxmrs** [comment]
@codex may I have another review?

**Comment by alxmrs** [comment]
@codex, may I have a review?

File: src/ocean_emulators/models/modules/blocks.py [hunk 1]
```diff
@@ -4,10 +4,33 @@
...
 from ocean_emulators.models.modules.activations import CappedGELU
 
 
+class PointwiseLinear(torch.nn.Module):
+    """A 1×1 convolution implemented as nn.Linear.
+
+    Mathematically equivalent to Conv2d(kernel_size=1), but avoids the
+    non-contiguous gradient strides that 1×1 convs produce, which cause
+    DDP to copy gradients instead of using zero-copy views.
+
```
Comment (jder) [inline src/ocean_emulators/models/modules/blocks.py:18]: @fomo-bot can you help me understand what "non-contiguous gradient strides" means in this context and what it has to do with DDP or copying/views? Do you think this will be a (time) performance improvement?

**Review by jder — APPROVED** [review]
